### DISCUSSION TITLE
Additional benchmarks?

### DISCUSSION THREAD

**Discussion by KristoferHansson** [eclipse-ecal/ecal#625] [General]
Hi,

Noticed eCAL v5.9.5 included performance fixes for when creating many publishers / consumers, I was wondering if you have any benchmark data available with regards to performance measurements / resource(cpu/mem) usage with different #consumers, #publishers, #topics, #publish frequency etc?

Recently did some tests on NATS and Zenoh focusing on these aspects since available data usually focuses on throughput or single producer/single consumer latency.

**Comment by FlorianReimold** [comment]
The performance fixes were actually about creating publishers and subscribers. During runtime nothing changed, only the instantiation of them.
But on 2022-05-09 we are going to release eCAL 5.10 which will come with some tweaks for shared memory communication (multi-buffering and zero copy). They have to be enabled manually as they are not compatible with 5.9 and below, but when you enable them they can lead to notably better performance in some scenarios!

**Reply by KristoferHansson** [reply]
Hi Florian

Thanks for the reply and the heads up regarding the shared memory improvements in 5.10.

Will probably await the 5.10 release and then run a similar set of measurements to compare runtime footprint given a set of varying load scenarios. If you already have something available possible to share then that would of course be of interest.

**Reply by FlorianReimold** [reply]
I think the only thing we currently have is this:
https://continental.github.io/ecal/advanced/performance.html

Maybe this helps you, but obviously performance measurements always depend on the hardware you are running on.

**Reply by KristoferHansson** [reply]
Yes, based my question on having read that. Also ran the sample benchmarking application with a few variations just to see what I would get with the current version running on my laptop in WSL2, compared to those benchmarks(from late 2020). Have not investigated further but was not able to get below 23us  for "Message average receive time". Will take a look at Rex ipc measurement setup mentioned below.

**Comment by rex-schilasky** [comment]
For 5.9 I made some latency measurements for local ipc, with and without zero copy, for windows and for linux. See [ecal_latency_ipc](https://github.com/rex-schilasky/ecal_latency_ipc). You could try to run that tests on your setup.

**Reply by KristoferHansson** [reply]
Thanks, will take a look!
